driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -361,6 +362,27 @@ private BsonDocument getCreateCollectionCommand() {
         return document;
     }
 
+    private void checkEncryptedFieldsSupported(final ConnectionDescription connectionDescription) throws MongoException {
+        if (encryptedFields != null && serverIsLessThanVersionSevenDotZero(connectionDescription)) {
+            throw new MongoClientException("Driver support of Queryable Encryption is incompatible with server."
+                    + " Upgrade server to use Queryable Encryption.");
+        }
+    }
+
+    /**
+     * @return {@code true} iff the {@linkplain #checkEncryptedFieldsSupported(ConnectionDescription) check} was successful,
+     * and the {@code callback} was not completed with a failed result.
+     */
+    private boolean checkEncryptedFieldsSupported(final AsyncConnection connection, final SingleResultCallback<Void> callback) {
```

Comment:
Consider changing the type of the first parameter to `ConnectionDescription` so it's clear there is no I/O being done.

Reply:
I am surprised I didn't do this originally.